fairlearn/datasets/_utils.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+
+# Common functions for retrieving ACS datasets as constructed in folktables:
+#    https://github.com/zykls/folktables
+
+# State Code based on 2010 Census definitions
+_STATE_CODES = {
+    "AL": "01",
+    "AK": "02",
+    "AZ": "04",
+    "AR": "05",
+    "CA": "06",
+    "CO": "08",
+    "CT": "09",
+    "DE": "10",
+    "FL": "12",
+    "GA": "13",
+    "HI": "15",
+    "ID": "16",
+    "IL": "17",
+    "IN": "18",
+    "IA": "19",
+    "KS": "20",
+    "KY": "21",
+    "LA": "22",
+    "ME": "23",
+    "MD": "24",
+    "MA": "25",
+    "MI": "26",
+    "MN": "27",
+    "MS": "28",
+    "MO": "29",
+    "MT": "30",
+    "NE": "31",
+    "NV": "32",
+    "NH": "33",
+    "NJ": "34",
+    "NM": "35",
+    "NY": "36",
+    "NC": "37",
+    "ND": "38",
+    "OH": "39",
+    "OK": "40",
+    "OR": "41",
+    "PA": "42",
+    "RI": "44",
+    "SC": "45",
+    "SD": "46",
+    "TN": "47",
+    "TX": "48",
+    "UT": "49",
+    "VT": "50",
+    "VA": "51",
+    "WA": "53",
+    "WV": "54",
+    "WI": "55",
+    "WY": "56",
+    "PR": "72",
+}
+
+
+class InvalidStateException(Exception):
+    """Exception thrown for invalid state codes."""
+
+    pass
+
+
+def check_states_valid(states):
```

Comment:
I am not sure why this should be included in Fairlearn - wouldn't it be easier to upload a version of the data set to OpenML that is already in the right format?

Reply:
Agreed. Does that mean I need to do it?

Reply:
I think I read this incorrectly. It seems to check whether the states supplied by the user (for filtering) are valid, not the dataset itself, which makes sense if we keep the `states` argument.